Go/ds/singly_linked_list.go

```diff
@@ -0,0 +1,174 @@
+/*
+	==============================================================
+		 Implementing Singly Linked List  data structure
+	==============================================================
+*/
+
+package main
+
+import (
+	"fmt"
+)
+
+// Node it is user-defined data type used to store data and the address of next node
+type Node struct {
+	next *Node
+	data int
+}
+
+// List it is user-defined data type used to represent the linked list
+type List struct {
+	head *Node
+	tail *Node
+	size int
+}
+
+// initList(): it will initalize the List's fields for first time
+func (list *List) initList() {
+	list.head = nil
+	list.tail = nil
+	list.size = 1
+}
+
+// isEmpty(): if the head of list is not poiting any node then it will return true otherwise false
+func (list *List) isEmpty() bool {
+	return list.head == nil
+}
+
+// insertAtFront(): if the list is empty then it will assign the same value to head and tail of linked list
+// otherwise it will insert node before the head in linked list
+func (list *List) insertAtFront() {
+	var value int
+	fmt.Println("enter the value of element:")
+	fmt.Scanf("%d", &value)
+	var node Node
+	node.data = value
+	if list.isEmpty() {
+		list.head = &node
+		list.tail = &node
+		list.size = 0
+	} else {
+		node.next = list.head
+		list.head = &node
+		list.size++
+	}
+	fmt.Println("value added successfully at the front of the linked list")
+}
+
+// insertAtEnd(): if the list is empty then it will assign the same value to head and tail of linked list
+// otherwise it will insert node after the tail in linked list
+func (list *List) insertAtEnd() {
+	var value int
+	fmt.Println("enter the value of element:")
+	fmt.Scanf("%d", &value)
+	var node Node
+	node.data = value
+	if list.isEmpty() {
+		list.head = &node
+		list.tail = &node
+		list.size = 0
+	} else {
+		list.tail.next = &node
+		list.tail = &node
+		list.size++
+	}
+	fmt.Println("value added successfully at the end of the linked list")
+}
+
+// insertAtGivenPosition(): if the list is empty then it will assign the same value to head and tail of linked list
+// otherwise it will insert node at the given psition in linked list
+func (list *List) insertAtGivenPosition() {
+	var position int
+	var value int
+	var node Node
+
+	fmt.Println("enter the position for element")
+	fmt.Scanf("%d", &position)
+	fmt.Println("enter the value of element:")
```

Comment:
Please change it to `enter the value of element to be inserted at above position:`